src/coreclr/src/vm/pefile.cpp

```diff
@@ -2142,8 +2142,15 @@ void PEAssembly::GetCodeBase(SString &result, BOOL fCopiedName/*=FALSE*/)
     }
     CONTRACTL_END;
 
-    // All other cases use the file path.
-    result.Set(GetEffectivePath());
+    if (GetILimage()->IsInBundle())
+    {
+        result.Set(SString::Empty());
```

Comment:
Basically here we would return null. Currently it looks weird that we make the same check twice in a row.

Reply:
I got lost here -- how would we return null? It looks like SString turns all null pointers into empty strings. Do I need to change the QCall signature?

Reply:
The QCall returns string, so that should not be a problem. `AssemblyNative::GetCodeBase` (Which is the QCall) should basically call `retString.Set(NULL)` - we will likely need to move the bundle logic into this method (or somewhat change the `PEAssembly::GetCodeBase` to be able to return null).

Reply:
I think the problem I'm hitting is https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/utilcode/sstring.cpp#L299-L300 where calling `result.Set(NULL)` ends up using an empty string.

Any suggestions?

Reply:
Do you actually need to handle the bundle case here? Can the entire fix be to just throws exception in `AssemblyNative::GetCodeBase`?

Reply:
This code path is also called by `GetAssemblyName`, and I think that should still work. I think it's fine if we set the `CodeBase` to `null` or empty string in the resulting `AssemblyName`, but I think `GetAssemblyName` is still a perfectly reasonable call.

Reply:
Ah, I see. I think what you have here is fine. I do not see a problem with IsInBundle being checked twice. Patterns like that happen all the time.

Reply:
Alternatively I could change the GetCodeBase call to return something other than `void` and use that as a sentinel for whether or not the app is in a bundle vs simply loaded from memory.

Personally I kind of like this approach, but if the pattern is frowned upon in the runtime code I'll keep the second QCall

Reply:
If it's too complicated I don't have a problem with the current implementation - it works. I was just trying to see if there's a simpler way...

Reply:
> I could change the GetCodeBase call to return something other than `void` and use that as a sentinel for whether or not the app is in a bundle vs simply loaded from memory.
> 
> Personally I kind of like this approach

I like it too, because it identifies the intention (ex: differentiates from retuning null/empty-string due to an unrelated bug).